crates/partition-store/src/persisted_lsn_tracking.rs

```diff
@@ -0,0 +1,261 @@
+use std::ffi::{CStr, CString};
+use std::sync::Arc;
+
+use ahash::HashMap;
+use dashmap::DashMap;
+use rocksdb::event_listener::{EventListener, FlushJobInfo};
+use rocksdb::table_properties::{
+    CollectorError, EntryType, TablePropertiesCollector, TablePropertiesCollectorFactory,
+};
+use tracing::warn;
+
+use restate_storage_api::StorageError;
+use restate_types::{identifiers::PartitionId, logs::Lsn};
+
+use crate::PARTITION_CF_PREFIX;
+use crate::fsm_table::{PartitionStateMachineKey, SequenceNumber, fsm_variable};
+use crate::keys::{KeyKind, TableKey};
+use crate::protobuf_types::PartitionStoreProtobufValue;
+
+pub(crate) struct LatestAppliedLsnCollector {
+    applied_lsns: HashMap<PartitionId, Lsn>,
+    properties: Vec<(CString, CString)>,
+}
```

Comment:
These are very short-lived - they get created during flush, and destroyed once done collecting properties.